Doesn't work with VST SDK 3.8.0

VST SDK 3.8.0 added forward-declared external C structs for the first time (for wayland support),

now this crate fails with errors like:

```
error[E0412]: cannot find type `xdg_toplevel` in this scope
    --> /Users/russruss/proj/sdks/vst3-rs/target/debug/build/vst3-df5410d317aec359/out/bindings.rs:1378:23
     |
1378 |             ) -> *mut xdg_toplevel
     |                       ^^^^^^^^^^^^ not found in this scope
```

This is because `xdg_toplevel` isn't defined so we don't parse the record definition.

--- a/com-scrape/src/clang.rs
+++ b/com-scrape/src/clang.rs
@@ -198,6 +198,15 @@ impl<'a> Cursor<'a> {
         unsafe { clang_equalCursors(self.cursor, clang_getCursorDefinition(self.cursor)) != 0 }
     }
 
+    pub fn has_definition(&self) -> bool {
+        unsafe {
+            clang_equalCursors(
+                clang_getNullCursor(),
+                clang_getCursorDefinition(self.cursor),
+            ) == 0
+        }
+    }
+
     pub fn type_(&self) -> Option<Type<'a>> {
         let type_ = unsafe { clang_getCursorType(self.cursor) };
         if type_.kind == CXType_Invalid {

--- a/com-scrape/src/parse.rs
+++ b/com-scrape/src/parse.rs
@@ -9,6 +9,7 @@ pub struct Namespace {
     pub children: BTreeMap<String, Namespace>,
     pub typedefs: Vec<Typedef>,
     pub records: Vec<Record>,
+    pub extern_records: Vec<ExternRecord>,
     pub constants: Vec<Constant>,
     pub unparsed_constants: Vec<String>,
 }
@@ -19,6 +20,7 @@ impl Namespace {
             children: BTreeMap::new(),
             typedefs: Vec::new(),
             records: Vec::new(),
+            extern_records: Vec::new(),
             constants: Vec::new(),
             unparsed_constants: Vec::new(),
         }
@@ -63,6 +65,11 @@ pub struct Record {
     pub bases: Vec<Base>,
     pub virtual_methods: Vec<Method>,
     pub inner: Namespace,
+}
+
+#[derive(Clone, Debug)]
+pub struct ExternRecord {
+    pub name: String,
 }
 
 #[derive(Clone, Debug)]
@@ -301,6 +308,10 @@ impl<'a> Parser<'a> {
                         let record = self.parse_record(cursor.type_().unwrap())?;
                         namespace.records.push(record);
                     }
+                } else if !cursor.has_definition() {
+                    namespace.extern_records.push(ExternRecord {
+                        name: cursor.name().to_str().unwrap().to_string(),
+                    });
                 }
             }
             _ => {}

--- a/com-scrape/src/print.rs
+++ b/com-scrape/src/print.rs
@@ -1,7 +1,7 @@
 use std::collections::HashSet;
 use std::io::{self, ErrorKind, Write};
 
-use crate::parse::{Method, Namespace, Record, RecordKind, Type, Value};
+use crate::parse::{ExternRecord, Method, Namespace, Record, RecordKind, Type, Value};
 use crate::Generator;
 
 struct UnnamedRecordScope {
@@ -72,6 +72,10 @@ impl<'a, W: Write> RustPrinter<'a, W> {
             self.print_record(&record)?;
         }
 
+        for record in &namespace.extern_records {
+            self.print_extern_record(&record)?;
+        }
+
         for constant in &namespace.constants {
             let name = &constant.name;
             write!(self.sink, "{indent}pub const {name}: ")?;
@@ -105,6 +109,19 @@ impl<'a, W: Write> RustPrinter<'a, W> {
 
         self.pop_unnamed_records()?;
 
+        Ok(())
+    }
+
+    fn print_extern_record(&mut self, record: &ExternRecord) -> io::Result<()> {
+        let indent = self.indent();
+        let name = &record.name;
+        writeln!(self.sink, "{indent}#[repr(C)]")?;
+        writeln!(self.sink, "{indent}#[derive(Copy, Clone)]")?;
+        writeln!(self.sink, "{indent}pub struct {name} {{")?;
+        writeln!(self.sink, "{indent}    _private: [u8; 0],")?;
+        writeln!(self.sink, "{indent}}}")?;
+        writeln!(self.sink, "{indent}unsafe impl Send for {name} {{}}")?;
+        writeln!(self.sink, "{indent}unsafe impl Sync for {name} {{}}")?;
         Ok(())
     }
 

